traits/tests/test_date.py

```diff
@@ -73,33 +73,45 @@ def test_assign_non_date(self):
         message = str(exception_context.exception)
         self.assertIn("must be a date or None, but", message)
 
-    def test_assign_datetime(self):
```

Comment:
This test is redundant with the new `test_assign_datetime_with_allow_datetime_not_given` test